### PR TITLE
RD-1723 - Adjust ExecutionProgress component to use new fields from executions API

### PR DESCRIPTION
**Pull request by qooban**
Recently Executions API was updated and it now provides two new fields:
* `total_operations`
* `finished_operations`
which can be used to calculate execution progress.

PR changelog:
* used new fields in `ExecutionProgress`
* removed workflow limitation for `ExecutionProgress` (now it works on all workflows)
* converted `ExecutionProgress` component to support TypeScript

Video:

https://user-images.githubusercontent.com/5202105/111434578-fd563580-86ff-11eb-9a95-feb33943401b.mp4

### CODE REVIEW

**Review by kubama — DISMISSED** [review]

File: widgets/deployments/src/ExecutionProgress.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        total_operations: number;
+        /* eslint-enable camelcase */
+    };
+}
+
+const ExecutionProgress: React.FunctionComponent<ExecutionProgressProps> = ({ execution }) => {
+    const { Progress } = Stage.Basic;
+    const { Execution } = Stage.Utils;
+
+    const color = Execution.isActiveExecution(execution) ? 'yellow' : 'green';
```
Comment (Gelio) [inline widgets/deployments/src/ExecutionProgress.tsx:15]: Optional:

To show that `yellow` and `green` are not arbitrary strings, we could do:

```suggestion
    const color: SemanticCOLORS = Execution.isActiveExecution(execution) ? 'yellow' : 'green';
```

which would require an `import type` from `semantic-ui-react'` :slightly_smiling_face:

File: widgets/deployments/src/ExecutionProgress.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    const error = Execution.isFailedExecution(execution);
+
+    const { finished_operations: finishedOperations, total_operations: totalOperations } = execution;
+    const ratio = finishedOperations / totalOperations;
+    const percent = Number.isFinite(ratio) ? Math.round(ratio * 100) : 0;
```
Comment (Gelio) [inline widgets/deployments/src/ExecutionProgress.tsx:20]: Looks like I will need a very similar operation in Deployments View.

Optional:

Would you mind extracting that to `DeploymentActions` or some other file?

I may also do it in my PR :slightly_smiling_face:

**Review by Gelio — APPROVED** [review]
Sad no file rename detection :worried: I suggest separating the functional changes from the changes related to TS into separate commits, so it's easier to review the functional changes :slightly_smiling_face: 

I appreciate the TS file :smile: Good work!

File: widgets/deployments/src/ExecutionProgress.tsx (hunk 2 above)
Comment (qooban) [reply widgets/deployments/src/ExecutionProgress.tsx:20]: Sure, good point. Added `getProgress` to `ExecutionUtils` as I think that's the right place for it.

File: widgets/deployments/src/ExecutionProgress.tsx (hunk 1 above)
Comment (qooban) [reply widgets/deployments/src/ExecutionProgress.tsx]: Sure, make sense, added 👍🏻

**Comment by qooban** [comment]
> Sad no file rename detection worried I suggest separating the functional changes from the changes related to TS into separate commits, so it's easier to review the functional changes slightly_smiling_face

In this case, I thought that it is such small that it'll not be a big problem in review.
But yeah, I'll definitely do that next time.

File: widgets/deployments/src/ExecutionProgress.tsx (hunk 2 above)
Comment (Gelio) [reply widgets/deployments/src/ExecutionProgress.tsx]: Sounds good :+1:

**Review by Gelio — APPROVED** [review]
:ok_hand:

**Pull request merged**